src/molecule/util.py

```diff
@@ -48,6 +48,10 @@
     from typing import Any, AnyStr, NoReturn
     from warnings import WarningMessage
 
+    Basic = str | float | bool
+    NestedDict = MutableMapping[str, Basic | None | list[Basic | "NestedDict"] | "NestedDict"]
+    _T = TypeVar("_T", bound=NestedDict)
```

Comment:
This also works, I just like having a little more specificity on what's actually coming out of these methods, and there's a lot of things that realistically aren't going to be in one of these dictionaries.

```suggestion
    _T = TypeVar("_T", bound=MutableMapping[str, Any])
```

Reply:
Take a look at https://github.com/ansible/ansible-compat/blob/main/src/ansible_compat/types.py